packages/zoltan2/core/src/input/Zoltan2_Adapter.hpp

```diff
@@ -243,11 +243,31 @@ template <typename User>
   // This function does not write edge info to the graph file, as the
   // BaseAdapter does not know about edge info; it writes
   // only the Chaco header and vertex weights (if applicable).
-  void generateWeightFileOnly(const char* fileprefix, 
+  void generateWeightFileOnly(const char* fileprefix,
                               const Teuchos::Comm<int> &comm) const;
 
 };
 
+template <typename User>
+class AdapterWithCoords : public BaseAdapter<User>
+{
+public:
+  virtual void getCoordinatesKokkosView(
+    Kokkos::View<typename BaseAdapter<User>::scalar_t **, Kokkos::LayoutLeft, typename BaseAdapter<User>::node_t::device_type> &elements) const = 0;
+};
+
+// Forward declare
+template <typename User>
+class VectorAdapter;
+
+template <typename User, typename UserCoord=User>
+class WrapperForCoords : public BaseAdapter<User>
```

Comment:
Perhaps "AdapterWithCoordsWrapper" would be a clearer name.
Will this inheritance work if the user does NOT have register coordinates with, say, a GraphAdapter?  That is, say a user wants to provide only Graph information, without coordinate information.  Will this code still work?